include/psa/crypto_struct.h

```diff
@@ -254,6 +254,18 @@ struct psa_key_attributes_s {
 #if defined(MBEDTLS_PSA_CRYPTO_SE_C)
     psa_key_slot_number_t MBEDTLS_PRIVATE(slot_number);
 #endif /* MBEDTLS_PSA_CRYPTO_SE_C */
+    /* Unlike normal buffers, there are three cases for domain_parameters
+     * and domain_parameters_size:
+     * - domain_parameters_size == SIZE_MAX && domain_parameters == NULL:
+     *   Access to domain parameters is not supported for this key.
+     *   This is a hack which should not exist, intended for keys managed
+     *   by a driver that doesn't support domain parameters.
```

Comment:
Minor: the wording suggests that there are two kinds of drivers: those that support domain parameters and those that don't. My understanding is that no drive can support domain parameters because there's no driver API for that (not defined in the spec, nor implemented in our code).